management/virtualmachine/entities.go

```diff
@@ -178,14 +287,24 @@ type RoleSize struct {
 	VirtualMachineResourceDiskSizeInMb int
 }
 
-type VirtualIPs struct {
-	VirtualIP []VirtualIP
+// Contains the definition of a DNS server for virtual machine deployment
+type DnsServer struct {
+	Name    string
+	Address string
 }
 
-type VirtualIP struct {
-	Address string
+// Contains the definition of a load balancer for virtual machine deployment
+type LoadBalancer struct {
+	Name                          string        // Specifies the name of the internal load balancer.
+	Type                          IPAddressType `xml:"FrontendIpConfiguration>Type"`                                    // Specifies the type of virtual IP address that is provided by the load balancer. The only allowable value is Private.
+	SubnetName                    string        `xml:"FrontendIpConfiguration>SubnetName,omitempty"`                    // Required if the deployment exists in a virtual network and a StaticVirtualNetworkIPAddress is assigned. Specifies the subnet of the virtual network that the load balancer uses. The virtual IP address that is managed by the load balancer is contained in this subnet.
+	StaticVirtualNetworkIPAddress string        `xml:"FrontendIpConfiguration>StaticVirtualNetworkIPAddress,omitempty"` // Specifies a specific virtual IP address that the load balancer uses from the subnet in the virtual network.
 }
 
+type IPAddressType string // see "IPAddressType*"" constants
+
+const IPAddressTypePrivate = IPAddressType("Private") // Only allowed value (currently) for IPAddressType
```

Comment:
Use const(....) block and "const ( xxx IPAddrType = "foo" ) should be ok, i.e. no need cast type